pkg/sensors/tracing/policyfilter_test.go

```diff
@@ -0,0 +1,327 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package tracing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cilium/tetragon/pkg/api/tracingapi"
+	"github.com/cilium/tetragon/pkg/bpf"
+	tgcgroups "github.com/cilium/tetragon/pkg/cgroups"
+	"github.com/cilium/tetragon/pkg/config"
+	grpcexec "github.com/cilium/tetragon/pkg/grpc/exec"
+	"github.com/cilium/tetragon/pkg/grpc/tracing"
+	"github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
+	"github.com/cilium/tetragon/pkg/logger"
+	"github.com/cilium/tetragon/pkg/observer"
+	"github.com/cilium/tetragon/pkg/option"
+	"github.com/cilium/tetragon/pkg/policyfilter"
+	"github.com/cilium/tetragon/pkg/reader/notify"
+	"github.com/cilium/tetragon/pkg/sensors/base"
+	"github.com/cilium/tetragon/pkg/sensors/config/confmap"
+	testsensor "github.com/cilium/tetragon/pkg/sensors/test"
+	"github.com/cilium/tetragon/pkg/testutils"
+	"github.com/cilium/tetragon/pkg/testutils/perfring"
+	tus "github.com/cilium/tetragon/pkg/testutils/sensors"
+	"github.com/google/uuid"
+
+	"github.com/containerd/cgroups"
+	cgroupsv2 "github.com/containerd/cgroups/v2"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+func FindCgroupDir(dir string) (string, error) {
+	cgRoot, err := tgcgroups.HostCgroupRoot()
+	if err != nil {
+		return "", err
+	}
+
+	var ret string
+	found := errors.New("Found")
+	filepath.WalkDir(cgRoot, func(path string, dentry fs.DirEntry, err error) error {
+		if err != nil || !dentry.IsDir() {
+			return nil
+		}
+		name := dentry.Name()
+		if strings.Contains(name, dir) {
+			ret = dir
+			return found
+		}
+		return nil
+	})
+	return filepath.Join(cgRoot, ret), nil
+}
+
+func createCgroup(t *testing.T, dir string, pids ...uint64) policyfilter.CgroupID {
+	cgMode := cgroups.Mode()
+	var path string
+	if cgMode == cgroups.Unified {
+		cgroupFs := "/sys/fs/cgroup"
+		res := cgroupsv2.Resources{}
+		m, err := cgroupsv2.NewSystemd("/", dir, -1, &res)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			m.DeleteSystemd()
+		})
+		for _, pid := range pids {
+			err = m.AddProc(pid)
+			require.NoError(t, err)
+		}
+		//path, err = FindCgroupDir(dir)
```

Comment:
nit: maybe we can remove that comment.

Reply:
yap, and FindCgroupDir with it I think. Thanks!